client/src/modules/Schedule/components/SettingsPanel/SettingsPanel.tsx

```diff
@@ -1,46 +1,79 @@
-import { Button, Col, Row } from 'antd';
-import { PlusOutlined, CopyOutlined } from '@ant-design/icons';
+import { Button, Col, message, Row } from 'antd';
+import { PlusOutlined, CopyOutlined, CalendarOutlined } from '@ant-design/icons';
 import { CourseScheduleItemDtoTagEnum } from 'api';
 import { ScheduleSettings } from 'modules/Schedule/hooks/useScheduleSettings';
 import React, { useState } from 'react';
 import { ManageEventModalForm } from '../ManageEventModalForm';
 import { SettingsDrawer } from '../SettingsDrawer';
 import ManageCsvButtons from './ManageCsvButtons';
+import { useCopyToClipboard } from 'react-use';
 
 interface SettingsPanelProps {
   isCourseManager: boolean;
   courseId: number;
+  courseAlias: string;
   settings: ScheduleSettings;
+  calendarToken: string;
   tags: CourseScheduleItemDtoTagEnum[];
   refreshData: () => void;
   onCreateCourseTask: () => void;
   onCopyFromCourse: () => void;
 }
 
+function buildICalendarLink(courseId: number, token: string, timezone: string) {
+  return `/api/v2/courses/${courseId}/icalendar/${token}?timezone=${encodeURIComponent(timezone || '')}`;
```

Comment:
btw, do I understand this right that using this token ical will be auto-updated?

Reply:
true